diff --git a/src/render/vulkan/SDL_render_vulkan.c b/src/render/vulkan/SDL_render_vulkan.c
--- a/src/render/vulkan/SDL_render_vulkan.c
+++ b/src/render/vulkan/SDL_render_vulkan.c
@@ -3738,7 +3738,7 @@ static VkSampler VULKAN_GetSampler(VULKAN_RenderData *data, SDL_ScaleMode scale_
         VkResult result = vkCreateSampler(data->device, &samplerCreateInfo, NULL, &data->samplerskey);
         if (result != VK_SUCCESS) {
             SET_ERROR_CODE("vkCreateSampler()", result);
-            return false;
+            return VK_NULL_HANDLE;
         }
     }
     return data->samplerskey;
